query-cloudtrail.sh

```diff
@@ -2,13 +2,24 @@
 
 set -euo pipefail
 
-JQ_TRANSFORM=".Events[] | [{EventId: .EventId, EventName: .EventName, EventTime: .EventTime, Username: .Username}]"
+JQ_QUERY="
+  .Events[] |
+    [{EventId: .EventId,
+      EventName: .EventName,
+      EventTime: .EventTime,
+      Username: .Username}]"
+
+# Get outputs from terraform state file
+state_value() {
+  key="$1"
+  yq read terraform-yaml/state.yml "terraform_outputs.$key"
```

Comment:
I know jq requires variables to be read ahead of time as arguments, does yq also need that or does the bash rendering happen first?

Reply:
You mean for the `"terraform_outputs.$key"` bit?  Since it's double-quotes, Bash will expand `$key` before `yq` sees the parameter.